### PR TITLE
Support svr host

### PR DESCRIPTION
**Pull request by ocalvo**
Fixes [#1823](https://github.com/home-assistant/addons/issues/1823)

### CODE REVIEW

**Comment by ocalvo** [comment]
Can you point me to the documentation on how to test this add-on before sending to production?

**Comment by ocalvo** [comment]
> Can you point me to the documentation on how to test this add-on before sending to production?

Found it: https://developers.home-assistant.io/docs/add-ons/testing

**Comment by ocalvo** [comment]
I cannot get the container to stop, it complains about this:
```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
s6-supervise (child): fatal: unable to exec run: No such file or directory
s6-supervise dnsmasqd: warning: unable to spawn ./run - waiting 10 seconds
[services.d] done.
s6-supervise (child): fatal: unable to exec run: No such file or directory
s6-supervise dnsmasqd: warning: unable to spawn ./run - waiting 10 seconds
s6-supervise (child): fatal: unable to exec run: No such file or directory
s6-supervise dnsmasqd: warning: unable to spawn ./run - waiting 10 seconds
s6-supervise (child): fatal: unable to exec run: No such file or directory
s6-supervise dnsmasqd: warning: unable to spawn ./run - waiting 10 seconds
s6-supervise (child): fatal: unable to exec run: No such file or directory
s6-supervise dnsmasqd: warning: unable to spawn ./run - waiting 10 seconds
```

**Comment by ocalvo** [comment]
I cannot get bashio to work:
```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] config.sh: executing...
: No such file or directory /usr/lib/bashio/bashio
[cont-init.d] config.sh: exited 127.
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.
```

**Comment by ocalvo** [comment]
Validated addon:
```
  docker run --rm -v w:\data:/data -p 53 local/dnsmasq_base
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] config.sh: executing...
[02:23:35] INFO: Configuring dnsmasq...
[cont-init.d] config.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[02:23:35] INFO: Starting dnsmasq...
dnsmasq[204]: started, version 2.84rc2 cachesize 150
dnsmasq[204]: compile time options: IPv6 GNU-getopt no-DBus no-UBus no-i18n no-IDN DHCP DHCPv6 no-Lua TFTP no-conntrack ipset auth no-cryptohash no-DNSSEC loop-detect inotify dumpfile
dnsmasq[204]: using nameserver 192.168.1.19#54
dnsmasq[204]: cleared cache
dnsmasq[204]: query[A] bing.com from 127.0.0.1
dnsmasq[204]: forwarded bing.com to 192.168.1.19
dnsmasq[204]: reply bing.com is 204.79.197.200
dnsmasq[204]: reply bing.com is 13.107.21.200
dnsmasq[204]: query[AAAA] bing.com from 127.0.0.1
dnsmasq[204]: forwarded bing.com to 192.168.1.19
dnsmasq[204]: reply bing.com is 2620:1ec:c11::200
dnsmasq[204]: cached _ldap._tcp.pdc._msdcs.mydomain.io is NODATA-IPv6
dnsmasq[204]: query[SRV] _ldap._tcp.pdc._msdcs.mydomain.io from 127.0.0.1
dnsmasq[204]: config _ldap._tcp.pdc._msdcs.mydomain.io is <SRV>
```

File: dnsmasq/rootfs/etc/cont-init.d/config.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# DNSMASQ config
+# ==============================================================================
+
+CONFIG="/etc/dnsmasq.conf"
+bashio::log.info "Configuring dnsmasq..."
+tempio \
+    -conf /data/options.json \
+    -template /usr/share/tempio/dnsmasq.config \
+    -out "${CONFIG}"
+
```
Comment (pvizeli) [inline dnsmasq/rootfs/etc/cont-init.d/config.sh:12]: As it is the main job, it need also a finish script

File: dnsmasq/config.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "Dnsmasq",
-  "version": "1.3",
+  "version": "1.4",
```
Comment (pvizeli) [inline dnsmasq/config.json:3]: You need also add `init: false` if you now use s6-overlay

File: dnsmasq/apparmor.txt [hunk 3]
```diff
@@ -25,6 +25,11 @@ profile dnsmasq flags=(attach_disconnected,mediate_deleted) {
...
 
-  /run.sh rix,
-  /data/** r,
+  /etc/s6/** ix,
+  /run/s6/** ix,
+  /etc/services.d/** rwix,
+  /etc/cont-init.d/** rwix,
+  /etc/cont-finish.d/** rwix,
+
+  /run/** rwk,
```
Comment (pvizeli) [inline dnsmasq/apparmor.txt:34]: Make sure they cover this basic: https://developers.home-assistant.io/docs/add-ons/presentation#apparmor

File: dnsmasq/apparmor.txt [hunk 4]
```diff
@@ -25,11 +25,15 @@ profile dnsmasq flags=(attach_disconnected,mediate_deleted) {
   /dev/tty rw,
   /tmp/* rw,
 
+  # S6-Overlay
+  /bin/** ix,
```
Comment (pvizeli) [inline dnsmasq/apparmor.txt:29]: Need a bit more cleanp. Like `/bin/busybox ix,` would be covered also with `/bin/** ix` ect

**Comment by ocalvo** [comment]
@pvizeli still fails in production, please rollback. It works fine in my test environment.

**Comment by raiju** [comment]
It looks like this is triggering install issues. The failure happens here:
https://github.com/home-assistant/supervisor/blob/6f26536d97c61feeb47857f666b69244a12091a5/supervisor/utils/apparmor.py#L28, where a regex pulls out lines starting w/ `profile `, and hard-codes there can only be one.

**Comment by raiju** [comment]
If helpful I opened a pure revert PR here: https://github.com/home-assistant/addons/pull/1860

**Comment by arjanhs** [comment]
It does fails now indeed:
21-02-16 07:26:47 ERROR (MainThread) [supervisor.utils.apparmor] To many profiles inside file: {'/usr/sbin/dnsmasq', 'dnsmasq'}

**Pull request merged**